doc/macos.md

```diff
@@ -1,18 +1,9 @@
+NOTE: This document might be outdated. Keeping this for reference for the signing process.
```

Comment:
Kill it and move it to BUILDING.md

Reply:
I know, it's just that we are not signing the binary so far because it is not really required for local dev. So this is not necessary for users right now, however it might be if we decide to buy a certificate and sign releases.

Reply:
signing binary on OS X was mandatory to use debug, did that change?

Reply:
i agree: this should be moved into building or renamed to `macos-signing-process.md`

Reply:
The reason I'm not removing this page nor making it super good is that I'm still not sure if we need this or not, but the info here are still useful and I'd like to keep them until we figure out if we need this or not and if we do, in what form.

Apparently to debug you just need to use the right entitlements and since we are anyway using a self-signed certificate, it works also the ad-hoc signing, which is done by using `codesign -s -` instead of `codesign -s re.rizin.rizin`.